src/app/modals/signup-pop-up/signup-pop-up.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatDialog } from '@angular/material/dialog';
+import { AuthService } from 'src/app/shared-module/auth/auth.service';
+import { SignupRequest } from 'src/app/shared-module/types/signup-request';
+import { SignupResult } from 'src/app/shared-module/types/signup-result';
+
+import { MatchValidator } from 'src/app/shared-module/validation/match';
+import { LoginPopUpComponent } from '../login-pop-up/login-pop-up.component';
+
+@Component({
+  selector: 'app-signup-pop-up',
+  templateUrl: './signup-pop-up.component.html',
+  styleUrls: [
+    '../login-pop-up/login-pop-up.component.scss',
+    './signup-pop-up.component.scss',
+  ],
+})
+export class SignupPopUpComponent {
+  signupRequest = <SignupRequest>{};
+  signupResult?: SignupResult;
+
+  signUpForm: FormGroup = new FormGroup({
+    name: new FormControl('', {
+      validators: [Validators.required, Validators.pattern('^[A-Za-z ]+$')],
+    }),
+    email: new FormControl('', {
+      validators: [Validators.required, Validators.email],
+    }),
+    password: new FormControl('', [
+      Validators.required,
+      Validators.minLength(8),
+      MatchValidator.Match('confirmPassword', true),
+    ]),
+    confirmPassword: new FormControl('', [
+      Validators.required,
+      Validators.minLength(8),
+      MatchValidator.Match('password'),
+    ]),
+    agreement: new FormControl('', [Validators.requiredTrue]),
+  });
```

Comment:
Just a small remark: you may use FormBuilder to make this code cleaner. It can be injected via constructor and a form group can be created by the following pattern:

`this.formGroup = formBuilder.group({ agreement: ['', Validators.requiredTrue], });`